Detectors/TRD/reconstruction/include/TRDReconstruction/EventRecord.h

```diff
@@ -152,16 +152,23 @@ class EventStorage
     mTFStats.mMCMDigitsFound[hcid / 2 * constants::NROBC1 * constants::NMCMROB + rob * constants::NMCMROB + mcm] += count;
   }*/
 
-  void incParsingError(int error, int sm, int side, int stacklayer)
+  void incParsingError(int error, int hcid)
   {
+    bool detailGraphs = true;
+    if (hcid < 0)
+      detailGraphs = false;
+    unsigned int sm = HelperMethods::getSector(hcid / 2);
+    unsigned int stack = HelperMethods::getStack(hcid / 2);
+    unsigned int layer = HelperMethods::getLayer(hcid / 2);
+    unsigned int side = hcid % 2;
```

Comment:
probably all these calculations should only be done if `hcid>=0`

Reply:
actually dont need those, gone.